src/stdio_mgr/stdio_mgr.py

```diff
@@ -95,62 +101,32 @@ def getvalue(self):
             return self._stream.getvalue().decode(self.encoding)
 
 
-@attr.s(slots=False)
-class TeeStdin(TextIOWrapper):
+class Tee(TextIOWrapper):
```

Comment:
I would want (at least) two implementations of teeing, ultimately, in order to accurately mimic the CLI experience: tee-on-read for stdin and tee-on-write for stderr.  That way, everything that would get pushed to a real user's console, across `in`/`out`/`err`, would get stored in `out` in the same order that that user would see it in "real life".

I had originally only implemented `TeeStdin` b/c I didn't have specific need for teeing `stderr` in my initial use-cases for it.

Reply:
Sounds good, but beyond the scope of this.  Raised https://github.com/bskinn/stdio-mgr/issues/65 about a similar feature.

However it does suggest that `Tee` should be `_Tee`, and other parts of this WIP probably should be private (even if they are desirable in coala-utils; that shouldnt be a consideration here)

Reply:
&lt;nod&gt;, agreed, out of scope here. 

But yes, exactly, if you think it's not a crazy feature to have, then it would be good to keep that additional teeing eventuality in mind during separation of the mocking and teeing of `stdin`.